templates/pkg/crd_sdk.go.tpl

```diff
@@ -369,3 +370,66 @@ func (rm *resourceManager) setStatusDefaults (
 		ko.Status.Conditions = []*ackv1alpha1.Condition{}
 	}
 }
+
+// updateConditions returns updated resource, true; if conditions were updated
+// else it returns nil, false
+func (rm *resourceManager) updateConditions (
+	r *resource,
+	err error,
+) (*resource, bool) {
+	ko := r.ko.DeepCopy()
+	rm.setStatusDefaults(ko)
+
+	// Terminal condition
+	var terminalCondition *ackv1alpha1.Condition = nil
+	for _, condition := range ko.Status.Conditions {
+		if condition.Type == ackv1alpha1.ConditionTypeTerminal {
+			terminalCondition = condition
+			break
+		}
+	}
+	if terminalCondition == nil {
+		terminalCondition = &ackv1alpha1.Condition{
+			Type:   ackv1alpha1.ConditionTypeTerminal,
+		}
+		ko.Status.Conditions = append(ko.Status.Conditions, terminalCondition)
+	}
+	if terminalErr, ok := rm.terminalAWSError(err); ok {
+		terminalCondition.Status = corev1.ConditionTrue
+		errorMessage := terminalErr.Message()
+		terminalCondition.Message = &errorMessage
+	} else {
+		terminalCondition.Status = corev1.ConditionFalse
+		terminalCondition.Message = nil
+	}
+
+{{ if $updateConditionsCustomMethodName := .CRD.UpdateConditionsCustomMethodName }}
+	// custom update conditions
+	rm.{{ $updateConditionsCustomMethodName }}(ko, r, err)
+{{ end }}
+	return &resource{ko}, true
+}
+
+// terminalAWSError returns awserr, true; if the supplied error is an aws Error type
+// and if the exception indicates that it is a Terminal exception
+// 'Terminal' exception are specified in generator configuration
+func (rm *resourceManager) terminalAWSError(err error) (awserr.Error, bool) {
+	if err == nil {
+		return nil, false
+	}
+{{- $terminalCodesCount := len .CRD.TerminalExceptionCodes }}
+	{{ if $terminalCodesCount -}} awsErr {{- else -}} _ {{- end }}, ok := ackerr.AWSError(err)
+	if !ok {
+		return nil, false
+	}
+{{- if $terminalCodesCount }}
+{{- $terminalCodesMaxIndex := minus $terminalCodesCount 1 }}
+	if {{ range $index, $terminalCode := .CRD.TerminalExceptionCodes -}}
+		awsErr.Code() == "{{ $terminalCode }}"
+		{{- if eq ($index) ($terminalCodesMaxIndex) }} { {{ else }} || {{ end }}
+		{{ end -}}
+		return awsErr, true
+	}
+{{ end }}
+	return nil, false
+}
```

Comment:
I think we can clean up this `terminalAWSError` method in a few ways:

1) Only have it return a `bool` and call it `isTerminalError()`

2) You can remove the need for the `{{ minus }}` template function like so:

```
if err == nil {
    return false
}
awsErr, ok := ackerr.AWSError(err)
if !ok {
    return false
}
{{ if .CRD.TerminalExceptionCodes }}
switch awsErr.Code() {
    case {{- range $x, $terminalCode := .CRD.TerminalExceptionCodes -}}{{ if ne ( $x 0 ) }}  || "{{ end }}{{ $terminalCode }}"{{ end }}:
        return true
    default:
        return false
    }
{{- else }}
    return false
{{- end }}
}
```

Reply:
Updated the template.